src/main/java/com/aws/greengrass/clientdevices/auth/certificate/ServerCertificateGenerator.java

```diff
@@ -91,6 +91,6 @@ public synchronized void generateCertificate(Supplier<List<String>> connectivity
                 .kv("certExpiry", getExpiryTime())
                 .log("New server certificate generated");
 
-        callback.accept(certificate);
+        callback.accept(certificate, certificateStore.getCaCertificateChain());
```

Comment:
Here we are passing fresh values to the event when a ca gets generated, avoiding the issue I described above with the closure.